packages/amqp-coffee/src/lib/channel.ts

```diff
@@ -374,10 +379,12 @@ export abstract class Channel extends EventEmitter {
 
       case methods.channelClose.name: {
         const args = frame.args
-        this.connection.channelManager.channelClosed(channel)
-
         debug(1, () => ['Channel closed by server', args])
+
+        this.connection._sendMethod(this.channel, methods.channelCloseOk, {})
```

Comment:
probably needs to be moved after

`          this.callbackForMethod(closingMethodOk)(args) // this would be the error`